internal/store/db/mongo/models/storedobject.go

```diff
@@ -0,0 +1,123 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+// models describes the data types that will be used when storing export data in Mongo.
+package models
+
+import (
+	"errors"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/store/contracts"
+
+	"github.com/google/uuid"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// StoredObject is the atomic and most abstract description of what is collected by the export store system.
+type StoredObject struct {
+	// ObjectID uniquely identifies this object in Mongo
+	ObjectID primitive.ObjectID `bson:"_id"`
+
+	// UUID uniquely identifies this StoredObject
+	UUID string `bson:"uuid"`
+
+	// AppServiceKey identifies the app to which this data belongs.
+	AppServiceKey string `bson:"appServiceKey"`
+
+	// Payload is the data to be exported
+	Payload []byte `bson:"payload"`
+
+	// RetryCount is how many times this has tried to be exported
+	RetryCount int `bson:"retryCount"`
+
+	// PipelinePosition is where to pickup in the pipeline
+	PipelinePosition int `bson:"pipelinePosition"`
+
+	// Version is a hash of the functions to know if the pipeline has changed.
+	Version string `bson:"version"`
+
+	// CorrelationID is an identifier provided by EdgeX to track this record as it moves
+	CorrelationID string `bson:"correlationID"`
+
+	// EventID is used to identify an EdgeX event from the core services and mark it as pushed.
+	EventID string `bson:"eventID"`
+
+	// EventChecksum is used to identify CBOR encoded data from the core services and mark it as pushed.
+	EventChecksum string `bson:"eventChecksum"`
+}
+
+func (o *StoredObject) FromContract(c contracts.StoredObject) error {
+	var err error
+	o.ObjectID, o.UUID, err = FromContractId(c.ID)
+	if err != nil {
+		return err
+	}
+
+	o.AppServiceKey = c.AppServiceKey
+	o.Payload = c.Payload
+	o.RetryCount = c.RetryCount
+	o.PipelinePosition = c.PipelinePosition
+	o.Version = c.Version
+	o.CorrelationID = c.CorrelationID
+	o.EventID = c.EventID
+	o.EventChecksum = c.EventChecksum
+
+	return nil
+}
+
+func (o StoredObject) ToContract() contracts.StoredObject {
+	contract := contracts.NewStoredObject(
+		o.AppServiceKey,
+		o.Payload,
+		o.PipelinePosition,
+		o.Version)
+
+	contract.ID = ToContractId(o.ObjectID, o.UUID)
+	contract.RetryCount = o.RetryCount
+	contract.CorrelationID = o.CorrelationID
+	contract.EventID = o.EventID
+	contract.EventChecksum = o.EventChecksum
+
+	return contract
+}
+
+func FromContractId(id string) (primitive.ObjectID, string, error) {
+	// In this first case, UUID is empty so this must be an add.
+	// Generate new BSON/UUIDs
+	if id == "" {
+		return primitive.NewObjectID(), uuid.New().String(), nil
+	}
+
+	// In this case, we're dealing with an existing id
+	objID, err := primitive.ObjectIDFromHex(id)
```

Comment:
So when doing an Update, will this properly recreate the ObjectId currently in the DB?

Reply:
Yes.